plugins/browser-plugin-snowplow-ecommerce/src/ua/api.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  trackProductView,
+  trackProductListView,
+  trackProductListClick,
+  trackCheckoutStep,
+  trackTransaction,
+  trackAddToCart,
+  trackRemoveFromCart,
+  trackPromotionClick,
+  trackPromotionView,
+} from '../api';
+import { CheckoutStep } from '../types';
+import {
+  AddToCartEvent,
+  CheckoutStepEvent,
+  ProductDetailEvent,
+  ProductListClickEvent,
+  ProductListViewEvent,
+  PromoClickEvent,
+  PromoViewEvent,
+  PurchaseEvent,
+  RemoveFromCartEvent,
+  UAEnhancedEcommerceObject,
+} from './types';
+
+import { transformUAProductsToSPProducts, transformUAPromotionsToSPPromotions } from './utils';
+
+type Options = {
+  currency?: string;
+};
+
+const DEFAULT_PAYMENT_METHOD = 'unknown';
+
+export function trackEnhancedEcommercePromoView(ecommerce: UAEnhancedEcommerceObject & PromoViewEvent) {
+  const promotions = transformUAPromotionsToSPPromotions(ecommerce.promoView.promotions);
+  /* Enhanced Ecommerce allows multiple promotions to be sent on a single event. */
+  for (let i = 0; i < promotions.length; i++) {
+    trackPromotionView(promotions[i]);
+  }
+}
+
+export function trackEnhancedEcommercePromoClick(ecommerce: UAEnhancedEcommerceObject & PromoClickEvent) {
+  const [promotion] = transformUAPromotionsToSPPromotions(ecommerce.promoClick.promotions);
+  trackPromotionClick(promotion);
+}
+
+export function trackEnhancedEcommerceAddToCart(
+  ecommerce: UAEnhancedEcommerceObject & AddToCartEvent,
+  opts: Options & { finalCartValue: number }
+) {
+  const currency = (ecommerce.currencyCode || opts.currency)!;
```

Comment:
I think now that we have this opts arg for ergonomics, we can blame the user if it falls back incorrectly. So I think we're now safe to return to your old version with like `const DEFAULT_CURRENCY_CODE = "USD";` or whatever as a default and remove the `!`. `const currency = ecommerce.currencyCode || opts.currency || DEFAULT_CURRENCY_CODE;`

Reply:
Sure!

Reply:
Sounds good!

I feel like an official plugin for the official SDK should basically never send data to bad if we can help it, but I think since this is a quick migration path it's better to preserve the (lack of) data quality and it's acceptable in this case.